### PR TITLE
Adopt lazy loading for Pic component

### LINKED ISSUE #1559: Pic component: Adopt lazy loading
The `` tag inside could use the `lazy` attribute -- browser support seems good https://caniuse.com/?search=lazy

The height / width resolution _might_ then have to be moved accordingly.

### PR DESCRIPTION
**Pull request by kimberlyohq**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [X] Feature addition or enhancement
- [ ] Code maintenance
- [ ] Others, please explain:

Resolves #1559 

**Overview of changes:**
Use `loading=lazy` attribute to defer loading of images for Pic components

Image is only loaded lazily as seen in the video
https://user-images.githubusercontent.com/60393696/124390276-5bde0880-dd1d-11eb-9cc9-f972f4995ad0.mp4

**Anything you'd like to highlight / discuss:**
In the issue description, it was mentioned that the height/width resolution might have to be moved accordingly. I tried a few examples and it seems to work as normal on my side. Is this concern regarding the `computeWidth` function? From what I googled online, the images are loaded once it reaches calculated distance from the viewport. (Source: https://web.dev/browser-level-image-lazy-loading/)
**Testing instructions:**

**Proposed commit message: (wrap lines at 72 characters)**

Adopt lazy loading for Pic component

---

**Checklist:** :ballot_box_with_check:

- [X] Updated the documentation for feature additions and enhancements
- [X] Added tests for bug fixes or features
- [X] Linked all related issues
- [X] No blatantly unrelated changes
- [ ] Pinged someone for a review!

### CODE REVIEW

**Comment by ang-zeyu** [comment]
> Is this concern regarding the computeWidth function? From what I googled online, the images are loaded once it reaches calculated distance from the viewport. (Source: https://web.dev/browser-level-image-lazy-loading/)

Thanks for investigating this. My concern was indeed around whether `naturalWidth / Height` would be populated correctly or at all if using the lazy attribute. Good to hear its unfounded 😀 


Other than that, wondering if we should document that lazy loading is used by default and provide a simple prop to disable this behaviour. Sometimes it might be preferable to load all the images in the document in one go for better ux.

**Comment by kimberlyohq** [comment]
> Other than that, wondering if we should document that lazy loading is used by default and provide a simple prop to disable this behaviour. Sometimes it might be preferable to load all the images in the document in one go for better ux.

Sounds good to me! WDYT about adding a `eager` prop if user wants to load images eagerly?

**Comment by ang-zeyu** [comment]
sounds good!

File: docs/userGuide/syntax/pictures.mbdf [hunk 1]
```diff
@@ -18,6 +22,7 @@ alt | `string` | | **This must be specified.**<br>The alternative text of the im
 height | `string` | | The height of the image in pixels.
 src | `string` | | **This must be specified.**<br>The URL of the image.<br>The URL can be specified as absolute or relative references. More info in: _[Intra-Site Links]({{baseUrl}}/userGuide/formattingContents.html#intraSiteLinks)_
 width | `string` | | The width of the image in pixels.<br>If both width and height are specified, width takes priority over height. It is to maintain the image's aspect ratio.
+eager | `boolean` | false | Whether the image is lazy loaded.
```
Comment (ang-zeyu) [inline docs/userGuide/syntax/pictures.mbdf:25]: Believe you meant something like this -
```suggestion
eager | `boolean` | false | The `<picture>` component lazy loads its images by default.<br>If you want to eagerly load the images, simply specify this attribute.
```

**Review by ang-zeyu — COMMENTED** [review]
Lgtm otherwise! 👍

File: docs/userGuide/syntax/pictures.mbdf [hunk 2]
```diff
@@ -18,6 +22,7 @@ alt | `string` | | **This must be specified.**<br>The alternative text of the im
 height | `string` | | The height of the image in pixels.
 src | `string` | | **This must be specified.**<br>The URL of the image.<br>The URL can be specified as absolute or relative references. More info in: _[Intra-Site Links]({{baseUrl}}/userGuide/formattingContents.html#intraSiteLinks)_
 width | `string` | | The width of the image in pixels.<br>If both width and height are specified, width takes priority over height. It is to maintain the image's aspect ratio.
+eager | `boolean` | false | The `<picture>` component lazy loads its images by default.<br>If you want to eagerly load the images, simply specify this attribute.
```
Comment (ang-zeyu) [inline docs/userGuide/syntax/pictures.mbdf:25]:
```suggestion
eager | `boolean` | false | The `<pic>` component lazy loads its images by default.<br>If you want to eagerly load the images, simply specify this attribute.
```

Comment (ang-zeyu) [reply docs/userGuide/syntax/pictures.mbdf]: made one tiny edit

**Review by ang-zeyu — APPROVED** [review]

**Pull request merged**